### ISSUE TITLE
Massive UPS hit

### ISSUE THREAD

**Issue by MDSGeist** [FKuerten/factorio-tree-growth-natural-expansion#2]
hey, just wanna make sure your aware,  i loaded up a map i havent played in a month, and after updating mods, and im getting massive ups hit from the  tree expansion.  so i checked the ms time usage on my mods in game and this mod is taking like 80-100 ms to update,   and all other mods are under 1 ms.  i'd be happy to give more info/ am also open to help with settings adjust.  thx

**Comment by FKuerten** [comment]
Yes, I am aware about this for a while (it has been reported on the mod page before).
I am currently testing other algorithms to see if they are faster.

**Comment by FKuerten** [comment]
I released 2.3.0. Please check if this improves the situation.

**Comment by MDSGeist** [comment]
yes!   amasing!   i think i can even make them less likely to spawn now
too!!.  solid 59/59.  yeah, i had a very big ups hit before because i had
the trees growing fast and spreadibng fast and allowed them to spawn almost
ontop of eachother.  i now have a super dense forest around my base, I also
started with toxic forest mod.  thank god for the chainsaw.  still havent
gotten to bots with this marathon bob's angel's map.  very good!  thankyou
for the mod!!

Tyler

**Comment by MDSGeist** [comment]
ok, after letting the game run for a while i got an error.

Error while running event tree-growth-natural-expansion::on_tick (ID 0)
__tree-growth-natural-expansion__/control/expansion.lua:141: bad argument
#1 to 'raise_event' (number expected, got nil)
